api/pkg/service/resource/resource.go

```diff
@@ -17,98 +17,132 @@ package resource
+	res := make(map[string]interface{})
 
-	req := res.Request{
-		Db:  s.DB(ctx),
-		Log: s.Logger(ctx),
-		ID:  p.ID,
+	if p.Name != "" {
+		res["Name"] = fmt.Sprintf("&name=%s", p.Name)
 	}
 
-	versions, err := req.AllVersions()
-	if err != nil {
-		if err == res.FetchError {
-			return nil, resource.MakeInternalError(err)
+	if p.Kinds != nil {
+		res["Kinds"] = fmt.Sprintf("&kinds=%s", p.Kinds[0])
+		if len(p.Kinds) > 1 {
+			for i := 1; i < len(p.Kinds); i++ {
+				res["Kinds"] = fmt.Sprintf("%v%s", res["Kinds"], fmt.Sprintf("&kinds=%s", p.Kinds[i]))
+			}
 		}
-		if err == res.NotFoundError {
-			return nil, resource.MakeNotFound(err)
+	}
+
+	if p.Categories != nil {
+		res["Categories"] = fmt.Sprintf("&categories=%s", p.Categories[0])
+		if len(p.Categories) > 1 {
+			for i := 1; i < len(p.Categories); i++ {
+				res["Categories"] = fmt.Sprintf("%v%s", res["Categories"], fmt.Sprintf("&categories=%s", p.Categories[i]))
+			}
 		}
 	}
 
-	var rv resource.Versions
-	rv.Versions = []*resource.ResourceVersionData{}
-	for _, r := range versions {
-		rv.Versions = append(rv.Versions, minVersionInfo(r))
+	if p.Catalogs != nil {
+		res["Catalogs"] = fmt.Sprintf("&catalogs=%s", p.Catalogs[0])
+		if len(p.Catalogs) > 1 {
+			for i := 1; i < len(p.Catalogs); i++ {
+				res["Catalogs"] = fmt.Sprintf("%v%s", res["Catalogs"], fmt.Sprintf("&catalogs=%s", p.Catalogs[i]))
+			}
+		}
 	}
-	rv.Latest = minVersionInfo(versions[len(versions)-1])
 
-	return &resource.ResourceVersions{Data: &rv}, nil
-}
+	if p.Tags != nil {
+		res["Tags"] = fmt.Sprintf("&tags=%s", p.Tags[0])
+		if len(p.Tags) > 1 {
+			for i := 1; i < len(p.Tags); i++ {
+				res["Tags"] = fmt.Sprintf("%v%s", res["Tags"], fmt.Sprintf("&tags=%s", p.Tags[i]))
+			}
+		}
+	}
 
-func tinyVersionInfo(r model.ResourceVersion) *resource.ResourceVersionData {
+	if p.Platforms != nil {
+		res["Platforms"] = fmt.Sprintf("&platforms=%s", p.Platforms[0])
+		if len(p.Platforms) > 1 {
+			for i := 1; i < len(p.Platforms); i++ {
+				res["Platforms"] = fmt.Sprintf("%v%s", res["Platforms"], fmt.Sprintf("&platforms=%s", p.Platforms[i]))
+			}
+		}
+	}
 
-	res := &resource.ResourceVersionData{
-		ID:      r.ID,
-		Version: r.Version,
+	if p.Limit != 1000 {
+		res["Limit"] = fmt.Sprintf("&limit=%d", p.Limit)
 	}
 
-	return res
-}
+	if p.Match == "exact" {
+		res["Match"] = fmt.Sprintf("&match=%s", p.Match)
+	}
 
-// This functions finds the minimum version information of
-// the resource such as rawURL, webURL, platforms and HubURL
-func minVersionInfo(r model.ResourceVersion) *resource.ResourceVersionData {
-
-	res := tinyVersionInfo(r)
-	res.WebURL = r.URL
-	res.RawURL = getStringReplacer(r.URL, r.Resource.Catalog.Provider).Replace(r.URL)
-	platforms := []*resource.Platform{}
-	for _, platform := range r.Platforms {
-		platforms = append(platforms, &resource.Platform{
-			ID:   platform.ID,
-			Name: platform.Name,
-		})
+	for _, v := range res {
+		query = fmt.Sprintf("%s%v", query, v)
```

Comment:
we are using map for query params construction , query params order gets change sometime after redirection e.g, `/query?tags=aws&tags=acr&limit=1` --> `/v1/query?limit=1&tags=aws&tags=ecr`.
I don't think it could be any issue , wdyt?

Reply:
yeah I don't think it would be any issue since the response would be the same